src/back-end/lib/resources/emailNotifications.ts

```diff
@@ -0,0 +1,90 @@
+import * as crud from 'back-end/lib/crud';
+import * as db from 'back-end/lib/db';
+import { handleTermsUpdated } from 'back-end/lib/mailer/notifications/termsUpdated';
+import * as permissions from 'back-end/lib/permissions';
+import { basicResponse, makeJsonResponseBody, wrapRespond } from 'back-end/lib/server';
+import { SupportedRequestBodies, SupportedResponseBodies } from 'back-end/lib/types';
+import { getString } from 'shared/lib';
+import { invalid, valid } from 'shared/lib/http';
+import { CreateRequestBody as SharedCreateRequestBody, CreateValidationErrors } from 'shared/lib/resources/emailNotifications';
+import { AuthenticatedSession, Session } from 'shared/lib/resources/session';
+import { adt } from 'shared/lib/types';
+
+type CreateRequestBody = SharedCreateRequestBody | null;
+
+export interface ValidatedCreateRequestBody {
+  session: AuthenticatedSession;
+  body: SharedCreateRequestBody;
+}
+
+type Resource = crud.Resource<
+  SupportedRequestBodies,
+  SupportedResponseBodies,
+  CreateRequestBody,
+  ValidatedCreateRequestBody,
+  CreateValidationErrors,
+  null,
+  null,
+  null,
+  null,
+  null,
+  null,
+  null,
+  Session,
+  db.Connection
+>;
+
+const resource: Resource = {
+  routeNamespace: 'email-notifications',
+
+  create(connection) {
+    return {
+      async parseRequestBody(request) {
+        const body = request.body.tag === 'json' ? request.body.value : {};
+        const tag = getString(body, 'tag');
+        switch (tag) {
+          case 'updateTerms':
+            return adt('updateTerms');
+          default:
+            return null;
+        }
+      },
+
+      async validateRequestBody(request) {
+        if (!request.body) { return invalid({ emailNotification: adt('parseFailure' as const) }); }
+        if (!permissions.updateTermsNotification(request.session) || !permissions.isSignedIn(request.session)) {
+          return invalid({
+            permissions: [permissions.ERROR_MESSAGE]
+          });
+        }
+
+        switch (request.body.tag) {
+          case 'updateTerms':
+            return valid({
+              session: request.session,
+              body: adt('updateTerms' as const)
+            });
+          default:
+            return invalid({ emailNotification: adt('parseFailure' as const) });
+        }
+      },
+
+      respond: wrapRespond({
+        valid: (async request => {
+          const { session, body } = request.body;
+          switch (body.tag) {
+            case 'updateTerms':
+              handleTermsUpdated(connection);
```

Comment:
@sutherlanda Need to set `acceptedTermsAt` to `null` for all users too, so we can track which users haven't yet accepted the latest terms.

Reply:
@dhruvio I had assumed that would be done in whatever method we use to actually update the terms, and that this would just be for a notification.  Do we want that logic in the emailNotification resource, or would it better to have it in a separate resource that we use to update the fixed content? I'm fine with either way, just want to confirm.

Reply:
@dhruvio I've made this change in the emailNotifications resource for now so that it doesn't block you.